SU-SWS/ansible-playbooks
# SITES-947: Re-factor parsing of sites.json to get the numerical site_id value

# READY FOR REVIEW

# Summary
- Dramatically reduce the time and resources that it takes to parse `sites.json` to get the `site_id` value

# Needed By (Date)
- ASAP

# Criticality
- How critical is this PR on a 1-10 scale?  10/10
- Significantly affects migration time

# Steps to Test

1. Use [this inventory](https://github.com/SU-SWS/ansible-playbooks/files/2439710/PR62.txt)
2. Migrate to `dev`, `test`, and `prod` with `full-migration-playbook.yml`
3. Ensure that `setup-site` step is skipped if the site exists on the environment
4. Ensure that caches are cleared and vhosts/custom domains are added
5. Delete one or both of the sites on one environment
6. Ensure that the site is created successfully
7. Migrate to one environment using `migration-sync-only-playbook.yml`
8. Run `post-db-restore-playbook.yml`
9. Run `add-vhost-playbook.yml`
10. Try to think of any edge cases that this approach would **not** work on

# Affected Projects or Products
- ACSF

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-947
- Any other contextual information that might be helpful:
    - I believe that this approach should be sufficient to get us a unique `site_id`. See comments in the `sites-json` role.
    - There is probably a more elegant way to do this leveraging Ansible (rather than shell commands like `grep` and `sed`, but this is a significant performance improvement, particularly when `sites.json` gets to be very large when there are many sites on a Factory.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
